packages/connect-query/package.json

```diff
@@ -1,21 +1,25 @@
 {
   "name": "@bufbuild/connect-query",
   "version": "0.2.0",
-  "main": "dist/src/index.js",
-  "types": "dist/src/index.d.ts",
+  "main": "dist/cjs/src/index.js",
+  "types": "dist/types/src/index.d.ts",
+  "exports": {
+    "import": "./dist/esm/src/index.js",
+    "require": "./dist/cjs/src/index.js",
+    "types": "./dist/types/src/index.d.ts"
+  },
   "license": "Apache-2.0",
   "sideEffects": false,
-  "files": [
-    "dist/**"
-  ],
   "repository": {
     "type": "git",
     "url": "https://github.com/bufbuild/connect-query.git",
     "directory": "packages/connect-query"
   },
   "scripts": {
     "clean": "rm -rf dist",
-    "build": "pnpm run clean && tsc --project tsconfig.build.json",
+    "build:cjs": "tsc --project tsconfig.build.cjs.json",
+    "build:esm": "tsc --project tsconfig.build.esm.json",
+    "build": "pnpm run clean && pnpm run build:cjs && pnpm run build:esm && tsx scripts/post-build.ts",
```

Comment:
Nit: I'd argue that post-build.ts should be run in the build:esm script.

Reply:
did you see the [other one](https://github.com/bufbuild/connect-query/blob/57bc26898734056b14193d0b3a96075909fefc18/packages/protoc-gen-connect-query/scripts/post-build.ts#L30)?  Indeed, I did originally do it exactly as you suggest (`post-build-cjs.ts`), but then the other package needed to also make a change to the `esm` so I made both `post-build` in the name of consistency.

whatddya think?

Reply:
(That's maintained by one of the TS maintainers: https://github.com/arethetypeswrong/arethetypeswrong.github.io)

Reply:
Hehe, that's awesome @fubhy 😆

Reply:
The nit was just about moving the call to the npm script @dimitropoulos, so that makes sense on it's own. 

I did not register the two scripts are different. But wait... Your package is CommonJS, so you need to put a `type:module` into dist/esm. 

Would be interesting to see if `arethetypeswrong` registers the package.json.